pkg/app/app.go

```diff
@@ -55,18 +56,32 @@ func NewApp(config *config.AppConfig) (*App, error) {
 }
 
 func (app *App) Run() error {
+	err := waitForTerminalSpace()
+	if err != nil {
+		return err
+	}
+	err = app.Gui.RunWithSubprocesses()
+	return err
+}
+
+func waitForTerminalSpace() error {
 	// before we do anything, we need to check that we have some window space available
-	var err error
-	width := 0
-	for width == 0 {
-		width, _, err = terminal.GetSize(int(os.Stdin.Fd()))
+	tStart := time.Now()
+	count := 0
+	for {
+		width, _, err := terminal.GetSize(int(os.Stdin.Fd()))
 		if err != nil {
 			return err
 		}
-		time.Sleep(time.Millisecond * 50)
+		if width != 0 {
+			return nil
+		}
+		count++
+		time.Sleep(time.Millisecond * 50 * time.Duration(count))
+		if count > 1 {
+			fmt.Fprintf(os.Stderr, "waited %s for available terminal space\n", time.Since(tStart))
```

Comment:
This message is not very friendly for someone who might not know what exactly "terminal space" means. Also, the first letter of the first word should be in the upper case.

Reply:
I actually wonder if we need to print this more than once. At 50 milliseconds we'll get 20 of these lines per second, which if we're for some reason waiting a while, will be printing way more lines than we need. I don't actually see an issue with just printing once 'Waiting for available window space to render panels" after the width comes back as 0 the first time.

Reply:
If we are going to print it once then we must not include the wait duration in the message.

Reply:
Yes I agree with all your points. I will simply do a print to stderr if the terminal space has zero width after 50 milliseconds (which should not happen)

Reply:
sounds good to me :)

Reply:
I propose this solution:
```diff
diff --git a/pkg/app/app.go b/pkg/app/app.go
index 4f84543..49d6478 100644
--- a/pkg/app/app.go
+++ b/pkg/app/app.go
@@ -4,7 +4,9 @@ import (
 	"fmt"
 	"io"
 	"os"
+	"os/signal"
 	"strings"
+	"syscall"
 	"time"
 
 	"golang.org/x/crypto/ssh/terminal"
@@ -66,21 +68,23 @@ func (app *App) Run() error {
 
 func waitForTerminalSpace() error {
 	// before we do anything, we need to check that we have some window space available
-	tStart := time.Now()
-	count := 0
-	for {
-		width, _, err := terminal.GetSize(int(os.Stdin.Fd()))
-		if err != nil {
-			return err
-		}
-		if width != 0 {
-			return nil
-		}
-		count++
-		time.Sleep(time.Millisecond * 50 * time.Duration(count))
-		if count > 1 {
-			fmt.Fprintf(os.Stderr, "waited %s for available terminal space\n", time.Since(tStart))
-		}
+	width, height, err := terminal.GetSize(int(os.Stdin.Fd()))
+	if err != nil {
+		return err
+	}
+	if width != 0 && height != 0 {
+		return nil
+	}
+
+	winch := make(chan os.Signal, 1)
+	signal.Notify(winch, syscall.SIGWINCH)
+	timeout := time.Second * 5
+
+	select {
+	case <-winch:
+		return nil
+	case <-time.After(timeout):
+		return fmt.Errorf("waited %s for available terminal space\n", timeout)
 	}
 }
 ```

Reply:
That's an elegant solution, adopting it ! But I think a delay of 1 second should be enough for any device, instead of 5. Also, is `syscall.SIGWINCH` working across all platforms? I was thinking to add a `terminal.GetSize` check again after the timeout in case SIGWINCH did not trigger. But if it works everywhere, then I'm happy as it is now.

Reply:
Looking at `go` source code it is present for Darwin+Linux on all architectures that we release `lazydocker` for.

Reply:
SIGWINCH is not present on windows tho, but seems like @jesseduffield is looking to support only WSL here https://github.com/jesseduffield/lazydocker/issues/29 and I'm in favor of this decision.

Reply:
Yes anyway the terminal size issue should only occur in Docker, which is running in Linux, so it should be fine.